### PR TITLE
Name validation

### PR DESCRIPTION
**Pull request by ddelpiano**
- name validation feature
- fixed a bug triggered by the delete of a component, the render was not triggered when required (introduced in the names_collision branch, PR https://github.com/MetaCell/geppetto-netpyne/pull/75 ).
- Dialog now showing up in case of name collision name validation and when the user insert a wrong variable name from the backend.

### CODE REVIEW

**Review by adrianq — CHANGES_REQUESTED** [review]
@ddelpiano I have done some preliminary tests and the PR seems to work fine and painful to code :) I have updated this branch with development. It was a bit tricky so it would be good if you can have a look at it and double check I haven't screwed up anything.

I have created a branch named name_validation in NetPyNE_UI and casper tests are running again. You can see [here](https://travis-ci.org/MetaCell/NetPyNE-UI/jobs/445396482) that tests are not passing yet. It looks like tests expect the 'id' attribute to be something like 'Population 2' and this is not the case anymore (as you change it to be 'Population2'). Can you please fix the tests?

There is a minor bug if I load tut3 and change the dropdown on the second tab to 'Create and Simulate Network'. I can see these three js errors althought the app works fine:
screenshot from 2018-10-24 00-11-48

**Comment by ddelpiano** [comment]
@adrianq bug fixed and casper tests as well, just restarted travis, will keep you posted. Thx.

**Comment by ddelpiano** [comment]
@adrianq development has been merged, re-tested locally and everything passed but to be sure I re-launched the job on travis. Feel free to merge. Thanks.

**Review by adrianq — COMMENTED** [review]
@ddelpiano PR looks good. Just a few minor comments:
- I merged with latest from development and tests are passing now.
- It concerns me a bit the complexity of the logic and the code for the name validation. I don't want to start refactoring code here but it looks to me some bits of code can potentially be extracted to some utils methods or common components. What do you think? It looks like most of the new code is inside handleRenameChange and shouldComponentUpdate and there is some common features that can potentially be extracted. If it makes sense for you can you maybe create a card in trello for this refactor in the future?
- Can we remove 'that' from the code? New JS syntax remove the need to use 'that' instead of 'this' in most of the cases...
- I see you have added a few comments, in general please add as much comments as possible. I suggested you some missing comments but please double check if you would like to add any other comment
- Also, along the same lines, can you please change variable name from model2 to something more explicit? I am trying to make the code as clearer as possible so we can spot any bug in the future easily.

Let me know when you are done with the changes and I will merge this PR.

File: Utils.js [hunk 1]
```diff
@@ -111,6 +111,16 @@ const Utils = {
             })
     },
 
+    nameValidation(myString) {
```
[the remark this answers is not in the document]
Comment (adrianq) [reply Utils.js]: I have already added some comments myself https://github.com/MetaCell/geppetto-netpyne/pull/76/commits/8f9538b2975870712d0fe30ebd23bd2626f32f00

**Comment by adrianq** [comment]
@ddelpiano I am merging this. It works pretty well and tests are passing. I have seen some minor issues with the rename validation and the refresh, i.e.the error popup shows up for a second and then disappears. We can try to fix it in another PR if you also notice the same behaviour.

I have created this card https://trello.com/c/qkTFWWUM/329-extract-validation-error-dialog-to-a-component to refactor some redundant code. It should be easy and useful. We also talked about refactoring/reducing the code in componentDidUpdate and rename handlers but not sure what you have in mind...

**Review by adrianq — APPROVED** [review]

**Pull request merged**